app/qml/map/MapWrapper.qml

```diff
@@ -297,8 +298,45 @@ Item {
   PositionKit {
     id: _positionKit
 
+    Component.onCompleted: {
+      // load previously active position provider
+      let providerId = __appSettings.activePositionProviderId
+
+      if ( !providerId ) // nothing has been written to qsettings
+      {
+        if ( __androidUtils.isAndroid || __iosUtils.isIos )
+        {
+          _positionKit.positionProvider = _positionKit.constructProvider( "internal" )
+        }
+        else // desktop
+        {
+          _positionKit.positionProvider = _positionKit.constructProvider( "simulated" )
+        }
+      }
+      else if ( providerId === "internal" )
+      {
+        _positionKit.positionProvider = _positionKit.constructProvider( "internal" )
+      }
+      else if ( providerId === "simulated" )
+      {
+        _positionKit.positionProvider = _positionKit.constructProvider( "simulated" )
+      }
+      else
+      {
+        _positionKit.positionProvider = _positionKit.constructProvider( "external", providerId )
+      }
```

Comment:
Maybe this logic could be moved to c++ and we would only keep something like this?
```qml
_positionKit.positionProvider = _positionKit.constructActiveProvider()
```